libnd4j/include/helpers/cpu/MmulHelper.cpp

```diff
@@ -164,7 +164,9 @@ static void usualDot(const Nd4jLong length, const double alpha, const void* vX,
     const bool betaPersent = beta;
 
     T3 sum = 0;
+#ifndef __NEC__
     PRAGMA_OMP_PARALLEL_FOR_ARGS(OMP_IF(length > Environment::getInstance().elementwiseThreshold()) schedule(guided) reduction(OMP_SUMT:sum))
+#endif
```

Comment:
Can't that go into the macro definition itself?

Reply:
@treo mind clarifying a bit? I just ifndefed anything NEC when it would break the compiler. @quickwritereader any suggestions on what you might do to this?

Reply:
he wants us to handle it inside include/system/openmp_pragmas.h

Reply:
@agibsonccc @quickwritereader has that exactly right. No point of ifdef-ing things at point of use, that is only going to bite you down the line.

https://github.com/eclipse/deeplearning4j/blob/d18b61aec46f89ce72dbac5dcf141e817ef4950f/libnd4j/include/system/openmp_pragmas.h#L26

Already has some handling of it, might as well put everything pragma related there.